SystemInformer/proctree.c

```diff
@@ -1926,33 +1931,17 @@ END_SORT_FUNCTION
 
 BEGIN_SORT_FUNCTION(Elevation)
 {
-    ULONG key1;
-    ULONG key2;
+    ULONG key1 = 0;
```

Comment:
nit: could be simplified using the fact that `ElevationType` is zero or greater, eliminating the extra branches.

```cpp
key1 = (processItem1->IsElevated ? 0 : 5) + processItem1->ElevationType;
key1 = (processItem1->IsElevated ? 0 : 5) + processItem1->ElevationType;
```

Reply:
Keys have following values:

- 0: if ElevationType = 0: possible value
- 1: No (Default): possible value
_- 2: No (Full): theoretical not possible_
- 3: No (Limited): possible value
- 6: Yes (Default): possible value
- 7: Yes (Full): : possible value
_- 8: Yes (Limited): theoretical not possible_

To reflect all possible combination, this is the easiest solution or i overlook something?

Reply:
Table looks fine. I'm just pointing out that you have an branching `if (processItem1->ElevationType)` and zero initialize of `keyX` that could be eliminated. I'm sure compiler optimizations will likely take care of it anyway.